greedy_max/__init__.py

```diff
@@ -293,66 +445,66 @@ def _insert(self, max_observation: Optional[Observation], time_window: Optional[
                         print('\nIntegrating max obs. over window...')
                         print('wstart', time_window.start)
                         print('wend', time_window.end)
-                        print('nttime', time_window.total_time)
-                        print('j values', np.arange(time_window.start, time_window.end - time_window.total_time + 2))
-
-                    for j in range(time_window.start, time_window.end - time_window.total_time + 2):
-                        f = sum(max_weights[j:j + time_window.total_time])
-
-                        # TODO: Error here: nttime is not defined. Is it time_window.total_time?
-                        # TODO: This will crash if verbose is on.
-                        # TODO: The only place I see nttime defined is in the _run method as a local variable.
-                        if self.verbose:
-                            print('j range', j, j + nttime - 1)
-                            print('obs weight', max_weights[j:j + nttime])
-                            print('integral', f)
-
-                        if f > maxf:
-                            maxf = f
-                            start = j
-                            end = start + time_window.total_time - 1
+                        print('nttime', time_window.time_slots)
+                        print('j values', np.arange(time_window.start, time_window.end - time_window.time_slots + 2))
+
+                    # Determine placement for maximum integrated weight 
+                    for window_idx in range(time_window.start, time_window.end - time_window.time_slots + 2):
```

Comment:
The range are the possible placements the observation can have inside the time window (ranging from `time_window.start` to `time_window.end`). I think `time_window.time_slots` might be the confusing part because this is actually the amount of time slots needed for the unit/observation to be schedule, so is equal in this case to `unit.length()` and so it might depict better the calculation.